src/components/Order/index.tsx

```diff
@@ -13,9 +17,32 @@ export const OrderDisplay = ({
   admin?: boolean;
 }) => {
   const [isFulfilled, setIsFulfilled] = useState(false);
+  const [{orderInfo}, dispatch] = useStateValue();
+  const [donutsOrdered, setDonutsOrdered] = useState<Donut[]>([]);
 
-  const handleFulfillOrder = () => {
+  const { _id, donuts: donutIds, drone, status } = order;
+  const abrevId = _id.slice(-3);
+
+  useEffect(() => {
+    const fetchDonuts = async () => {
+      const donuts: Donut[] = [];
+
+      for (const donutId of donutIds) {
+        const response = await fetch(`${ServerUrl}/api/donut/${donutId}`);
```

Comment:
I saw lots of errors here since donutId is an object instead of a string. You probably needed a stringify somewhere but I took the chance to clean up the code a bit here.